Cubical/Categories/Instances/FPAlgebrasSmall.agda

```diff
@@ -0,0 +1,334 @@
+open import Cubical.Algebra.CommAlgebra
+open import Cubical.Algebra.CommAlgebra.FPAlgebra
+open import Cubical.Algebra.CommAlgebra.FPAlgebra.Instances
+open import Cubical.Algebra.CommAlgebra.Instances.Unit
+open import Cubical.Algebra.Semilattice
+open import Cubical.Algebra.Lattice
+open import Cubical.Algebra.DistLattice
+open import Cubical.Algebra.DistLattice.BigOps
+open import Cubical.Algebra.ZariskiLattice.Base
+open import Cubical.Algebra.ZariskiLattice.UniversalProperty
+
+open import Cubical.Categories.Category
+open import Cubical.Categories.Functor
+open import Cubical.Categories.Instances.Sets
+open import Cubical.Categories.Instances.CommRings
+open import Cubical.Categories.Instances.CommAlgebras
+open import Cubical.Categories.Instances.DistLattices
+open import Cubical.Categories.Instances.Functors
+open import Cubical.Categories.NaturalTransformation
+open import Cubical.Categories.Presheaf.Base
+open import Cubical.Categories.Yoneda
+
+open import Cubical.Relation.Binary.Order.Poset
+
+open import Cubical.HITs.PropositionalTruncation as PT
+open import Cubical.HITs.SetQuotients as SQ
+
+
+private
+ variable
+  ℓ ℓ' ℓ'' : Level
+
+
+module _ (R : CommRing ℓ) where
+
+  open Iso
+  open Category hiding (_∘_) renaming (_⋆_ to _⋆c_)
+  open Functor
+  open NatTrans
+  open DistLatticeStr ⦃...⦄
+  open CommRingStr ⦃...⦄
+  open IsRingHom
+  open CommAlgebraHoms
+  open IsLatticeHom
+  open ZarLat
+  open ZarLatUniversalProp
+
+
+  -- type living in the same universe as the base ring R
+  record SmallFPAlgebra : Type ℓ where
+    field
+      n : ℕ
+      m : ℕ
+      relations : FinVec ⟨ Polynomials {R = R} n ⟩ m
+
+  open SmallFPAlgebra
+  private
+    indFPAlg : SmallFPAlgebra → CommAlgebra R ℓ
+    indFPAlg A = FPAlgebra (A .n) (A .relations)
+
+  FPAlgebrasSmallCategory : Category ℓ ℓ
+  ob FPAlgebrasSmallCategory = SmallFPAlgebra
+  Hom[_,_] FPAlgebrasSmallCategory A B =
+    CommAlgebraHom (indFPAlg A) (indFPAlg B)
+  id FPAlgebrasSmallCategory = idCommAlgebraHom _
+  _⋆c_ FPAlgebrasSmallCategory = compCommAlgebraHom _ _ _
+  ⋆IdL FPAlgebrasSmallCategory = compIdCommAlgebraHom
+  ⋆IdR FPAlgebrasSmallCategory = idCompCommAlgebraHom
+  ⋆Assoc FPAlgebrasSmallCategory = compAssocCommAlgebraHom
+  isSetHom FPAlgebrasSmallCategory = isSetAlgebraHom _ _
+
+
+  -- the name of this module will be justified once we have defined
+  -- Zariski sheaves on small f.p. algebras
+  module SchemesOfFinitePresentation where
+
+    -- "parsimonious" presheaf categories
+    fpAlgFunctor = Functor FPAlgebrasSmallCategory (SET ℓ)
+    fpAlgFUNCTOR = FUNCTOR FPAlgebrasSmallCategory (SET ℓ)
+
+    -- yoneda in the notation of Demazure-Gabriel
+    -- uses η-equality of Categories
+    Sp : Functor (FPAlgebrasSmallCategory ^op) fpAlgFUNCTOR
+    Sp = YO {C = (FPAlgebrasSmallCategory ^op)}
+
+    -- the Zariski lattice (classifying compact open subobjects)
+    private forget = ForgetfulCommAlgebra→CommRing R
+
+    𝓛 : fpAlgFunctor
```

Comment:
This should probably be defined by just composing the functor `forget` with the Z-functor version of `𝓛`.